src/tools/perf/perftest.c

```diff
@@ -790,6 +794,14 @@ static ucs_status_t parse_test_params(perftest_params_t *params, char opt,
             return UCS_OK;
         }
         return UCS_ERR_INVALID_PARAM;
+    case 'R':
+        if ((0.0 <= atof(opt_arg)) && (atof(opt_arg) <= 100.0)) {
```

Comment:
maybe set `params->super.pctl_rank = atof(opt_arg)` and then check it's valid value?

Reply:
Fixed